### DISCUSSION TITLE
47. 캡슐화와 역캡슐화가 무엇인지 설명해 주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#49] [General]
`키워드`: 캡슐화, 역캡슐화, 패킷 구조, 헤더, 트레일러

**Comment by cutehumanS2** [comment]
캡슐화와 역캡슐화는 네트워크에서 데이터를 전송하고 수신하는 과정에서 수행됩니다.

- **캡슐화**는 데이터를 상위 계층에서 하위 계층으로 송신할 때, 각 계층의 프로토콜에 맞게 헤더나 트레일러를 추가하는 과정입니다.
- **역캡슐화**는 데이터를 수신할 때, 헤더와 트레일러를 제거하여 원본 데이터를 상위 계층으로 전달하는 과정입니다.

**Comment by cutehumanS2** [comment]
**[캡슐화를 하는 목적이 무엇인가요?]**
- 데이터를 전송하기 위한 프로토콜 정보를 추가하고, 목적지까지 신뢰성있게 전달하기 위함입니다.
- 계층마다 자신의 역할에 맞게 데이터를 처리하고 헤더를 추가하므로, 각 계층의 독립성을 유지할 수 있습니다.
- 서로 다른 장치나 시스템에서도 규칙화된 데이터 전송 구조를 통해 호환성을 보장합니다.
<br>

**[역캡슐화를 하는 목적이 무엇인가요?]**
- 최종적으로 애플리케이션이 이해할 수 있는 데이터 형태로 변환하기 위함입니다.
- 역캡슐화 과정에서 각 계층의 헤더 정보를 확인하며, 데이터의 순서와 무결성 여부를 검증하기 위함입니다. 이를 통해 신뢰할 수 있는 데이터 전달을 보장합니다.
